doc/api/crypto.md

```diff
@@ -1046,6 +1046,76 @@ encoding of `'utf8'` is enforced. If `data` is a [`Buffer`][], `TypedArray`, or
 
 This can be called many times with new data as it is streamed.
 
+## Class: KeyObject
+<!-- YAML
+added: REPLACEME
+-->
+
+Node.js uses an internal `KeyObject` class which should not be accessed
+directly. Instead, factory functions exist to create instances of this class
+in a secure manner, see [`crypto.createSecretKey`][],
+[`crypto.createPublicKey`][] and [`crypto.createPrivateKey`][]. A `KeyObject`
+can represent a symmetric or asymmetric key, and each kind of key exposes
+different functions.
+
+Most applications should consider using the new `KeyObject` API instead of
+passing keys as strings or `Buffer`s due to improved security features.
+
+### keyObject.getType()
+<!-- YAML
+added: REPLACEME
+-->
+* Returns: {string}
+
+Depending on the type of this `KeyObject`, this function either returns
+`'secret'` for symmetric keys, `'public'` for public (asymmetric) keys or
+`'private'` for private (asymmetric) keys.
+
+### keyObject.getSymmetricSize()
+<!-- YAML
+added: REPLACEME
+-->
+* Returns: {number}
+
+For symmetric keys, this function returns the size of the key in bytes. This
+function is undefined for asymmetric keys.
+
+### keyObject.getAsymmetricKeyType()
+<!-- YAML
+added: REPLACEME
+-->
+* Returns: {string}
+
+For asymmetric keys, this function returns the type of the embedded key (e.g.,
+`'rsa'` or `'dsa'`). This function is undefined for symmetric keys.
+
+### keyObject.export([options])
+<!-- YAML
+added: REPLACEME
+-->
+* `options`: {Object}
+* Returns: {string | Buffer}
+
+For symmetric keys, this function allocates a `Buffer` containing the key
+material and ignores any options.
+
+For asymmetric keys, the `options` parameter is used to determine the export
+format. For public keys, the following encoding options can be used:
+
+* `type`: {string} Must be one of `'pkcs1'` (RSA only) or `'spki'`.
+* `format`: {string} Must be `'pem'` or `'der'`.
+
+For private keys, the following encoding options can be used:
+
+* `type`: {string} Must be one of `'pkcs1'` (RSA only), `'pkcs8'` or
+  `'sec1'` (EC only).
+* `format`: {string} Must be `'pem'` or `'der'`.
```

Comment:
no defaults? I'd say defaulting to pkcs8/pem would make sense in the node sense (with pkcs encrypted if a passphrase is provided)

Reply:
I originally avoided defaults when I implemented key encoding options in https://github.com/nodejs/node/pull/22660 to make future changes simpler, but at this point, we could definitely add defaults. PKCS8 and SPKI are obvious choices, and PEM seems reasonable as well. I tried to avoid that since WebCrypto uses DER by default, but PEM makes more sense in Node.js.

Reply:
I'm OK with no defaults for a while, as long as it throws when insufficient options are specified (so users don't think that there are defaults).